test/modules/DirectMethodSender/Program.cs

```diff
@@ -141,5 +110,31 @@ public static bool IsTestTimeUp(DateTime testStartAt)
         {
             return (Settings.Current.TestDuration == TimeSpan.Zero) || (DateTime.UtcNow - testStartAt < Settings.Current.TestDuration);
         }
+
+        // Create reporting result depending on which endpoint is being used.
+        public static TestResultBase ConstructTestResult(DirectMethodResultType directMethodResultType, Guid batchId, long counter, string result)
+        {
+            string source = Settings.Current.ModuleId + ".send";
+            switch (directMethodResultType)
+            {
+                case DirectMethodResultType.DirectMethodTestResult:
```

Comment:
We should have 3 different type for E2E, long haul and connectivity test.

Reply:
Nope. 
`DirectMethodTestResult `for TRC. 
`LegacyDirectMethodTestResult `for LH & E2E